test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetListCommandTest.cs

```diff
@@ -101,6 +101,132 @@ public void ListCommand_WithUserSpecifiedSource()
+                var packageFileNameA1 = Util.CreateTestPackage(packageAId, "1.0.1", packageDirectory);
+                var packageFileNameA2 = Util.CreateTestPackage(packageAId, "2.0.0", packageDirectory);
+                var packageFileNameB1 = Util.CreateTestPackage(packageBId, "1.1.0", packageDirectory);
+                var packageFileNameBPreview = Util.CreateTestPackage(packageBId, "9.2.0-Preview1", packageDirectory);
+
+                // Act
+                var args = $"list -source {packageDirectory} " +
+                            (includeMatchingText ? "best" : string.Empty) + // matching text
+                            (preRelease ? " -prerelease " : string.Empty);
+
+                var r1 = CommandRunner.Run(
+                    nugetexe,
+                    solutionFolder,
+                    args,
+                    waitForExit: true);
+
+                // Assert
+                Assert.Equal(0, r1.ExitCode);
+                var expectedOutput = $"{packageAId} 2.0.0" + Environment.NewLine +
+                                     $"{packageBId} " + (preRelease ? "9.2.0-Preview1" : "1.1.0") + Environment.NewLine;
+                r1.Output.Should().Be(expectedOutput);
+            }
+        }
+
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void ListCommand_WithLocalSource_WithNoMatchingText_ReturnEmpty(bool preRelease, bool allVersions)
+        {
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                var packageDirectory = Path.Combine(pathContext.WorkingDirectory, "packageFolder");
+                var solutionFolder = Path.Combine(pathContext.SolutionRoot);
+                string prefix = "newBestPackage";
+                string packageAId = prefix + "A";
+                string packageBId = prefix + "B";
+                var packageFileNameABeta = Util.CreateTestPackage(packageAId, "1.0.0-beta.1.2", packageDirectory);
+                var packageFileNameA1 = Util.CreateTestPackage(packageAId, "1.0.1", packageDirectory);
+                var packageFileNameA2 = Util.CreateTestPackage(packageAId, "2.0.0", packageDirectory);
+                var packageFileNameB1 = Util.CreateTestPackage(packageBId, "1.1.0", packageDirectory);
+                var packageFileNameBPreview = Util.CreateTestPackage(packageBId, "9.2.0-Preview1", packageDirectory);
+
+                // Act
+                var args = $"list -source {packageDirectory} " +
+                            "randomId" + // this Id not part of any package Id
+                            (preRelease ? " -prerelease " : string.Empty) +
+                            (allVersions ? " -allversions " : string.Empty);
```

Comment:
since `randomId` is not part of any package ID, that makes me feel like testing `preRelease` and `allVersions` as a test matrix isn't actually improving our test coverage.  If checking both prerelease and all versions doesn't list any output, then telling `list` to filter out (not show) pre release and/or all versions, it's going to have the same result.

Different tests already test that prerelease and all versions flags work, so it's not only duplicated tests, but this test wouldn't even catch a bug where either of those flags were not implemented correctly, since all the packages are being filtered out due to not matching the package id filter.

Reply:
Sure, I removed this slow tests.